c/meterpreter/source/extensions/stdapi/server/net/net.c

```diff
@@ -1 +1,72 @@
-#include "precomp.h"
+#include "precomp.h"
+
+#include <winsock2.h>
+#include <ws2tcpip.h>
+
+static char * parse_sockaddr(struct sockaddr_storage *addr, uint16_t *port)
+{
+	char *host = NULL;
+
+	host = malloc(INET6_ADDRSTRLEN);
+	if (host) {
+		if (addr->ss_family == AF_INET) {
+			struct sockaddr_in *s = (struct sockaddr_in *)addr;
+			*port = ntohs(s->sin_port);
+			inet_ntop(AF_INET, &s->sin_addr, host, INET6_ADDRSTRLEN);
+		}
+		else if (addr->ss_family == AF_INET6) {
+			struct sockaddr_in6 *s = (struct sockaddr_in6 *)addr;
+			*port = ntohs(s->sin6_port);
+			inet_ntop(AF_INET6, &s->sin6_addr, host, INET6_ADDRSTRLEN);
+		}
+	}
+	return host;
+}
+
+const char * inet_ntop(int af, const void *src, char *dst, socklen_t size) {

```

Comment:
I'm pretty sure defining this is necessary because there is not an implementation available using the older runtime that we're using for compatibility. This should also be used where (if I'm reading [this](https://github.com/rapid7/metasploit-payloads/blob/18ed237c1d9ae70030d1b01e64eb67b2c75fa9db/c/meterpreter/source/extensions/stdapi/server/net/socket/tcp_server.c#L222-L230) correctly) IPv6 addresses are returned as empty strings. That's a PR for another day.